subworkflows/local/utils_nfcore_eager_pipeline/main.nf

```diff
@@ -81,55 +81,82 @@ workflow PIPELINE_INITIALISATION {
     //
     // Create channel from input file provided through params.input
     //
-    ch_samplesheet_for_branch = Channel.fromSamplesheet("input")
-                                    .map{
+    ch_samplesheet = Channel.fromSamplesheet("input")
+                                    .map {
                                         meta, r1, r2, bam ->
                                             meta.single_end = meta.pairment == "single" ? true : false
+                                            meta.id = meta.sample_id
                                         [ meta, r1, r2, bam ]
                                     }
+
+    ch_samplesheet_for_branch = ch_samplesheet
                                     .branch {
                                         meta, r1, r2, bam ->
                                             bam: bam.toString().endsWith(".bam")
                                             fastq: true
                                     }
 
     ch_samplesheet_fastqs = ch_samplesheet_for_branch.fastq
-                                .map{
+                                .map {
                                     meta, r1, r2, bam ->
                                         reads = meta.single_end ? [ r1 ] : [ r1, r2 ]
                                     [ meta - meta.subMap('pairment', 'bam_reference_id'), reads ]
                                 }
 
     ch_samplesheet_bams = ch_samplesheet_for_branch.bam
-                            .map{
+                            .map {
                                 meta, r1, r2, bam ->
                                     meta.reference = meta.bam_reference_id
                                     meta.id_index = meta.bam_reference_id
-                                [ meta - meta.subMap('bam_reference_id'), bam ]
+                                [ meta - meta.subMap('pairment', 'bam_reference_id'), bam ]
```

Comment:
Do we really want to remove `bam_reference_id` here? I would assume that needs to be retained for joining with the corresponding reference

Reply:
All other parts of the pipeline use meta.reference or meta.id_index which are created above.